spec/unit/type/acl_spec.rb

```diff
@@ -684,35 +684,43 @@ def test_should_not_set_autorequired_file(resource_path,file_path)
       end
     end
 
-    context ":type" do
+    context ":perm_type" do
       it "should default to allow" do
         resource[:permissions] = {'identity' =>'bob','rights'=>['full']}
-        resource[:permissions][0].type.should == :allow
+        resource[:permissions][0].perm_type.should == :allow
       end
 
       it "should accept allow" do
-        resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'type'=>'allow'}
+        resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'perm_type'=>'allow'}
       end
 
       it "should accept deny" do
-        resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'type'=>'deny'}
+        resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'perm_type'=>'deny'}
       end
 
       it "should reject any other value" do
         expect {
-          resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'type'=>'what'}
+          resource[:permissions] = {'identity' =>'bob','rights'=>['full'],'perm_type'=>'what'}
         }.to raise_error(Puppet::ResourceError, /Invalid value "what". Valid values are/)
       end
 
       it "should reject empty" do
         expect {
-          resource[:permissions] = {'identity'=>'bob','rights'=>['full'],'type'=>''}
+          resource[:permissions] = {'identity'=>'bob','rights'=>['full'],'perm_type'=>''}
         }.to raise_error(Puppet::ResourceError, /Invalid value "". Valid values are/)
       end
 
       it "should set default value on nil" do
-        resource[:permissions] = {'identity'=>'bob','rights'=>['full'],'type'=>nil}
-        resource[:permissions][0].type.should == :allow
+        resource[:permissions] = {'identity'=>'bob','rights'=>['full'],'perm_type'=>nil}
+        resource[:permissions][0].perm_type.should == :allow
+      end
+
+      it "should munge `type` to `perm_type`" do
+        require 'pry'
+        resource[:permissions] = {'identity' => 'bob', 'rights' => ['full'], 'type' => 'deny'}
+        resource[:permissions][0].perm_type.should == :deny
+        expect(@logs[0].level).to equal(:warning)
+        @logs[0].message.should eq('Permission `type` is deprecated and has been replaced with perm_type for allow or deny')
```

Comment:
Okay, looks like you are doing that with this test. You need to split the test into two and test the items separately. I didn't notice that the contents of the test were doing so much because I read the description. Remember that the description should explain everything you are testing. If it is too long, you are testing too much and should split to multiple tests.